pkgs/build-support/fetchgit/default.nix

```diff
@@ -55,12 +55,14 @@ in
 */
 
 assert deepClone -> leaveDotGit;
-assert nonConeMode -> (sparseCheckout != "");
+assert nonConeMode -> (sparseCheckout != []);
 
 if md5 != "" then
   throw "fetchgit does not support md5 anymore, please use sha256"
 else if hash != "" && sha256 != "" then
   throw "Only one of sha256 or hash can be set"
+else if builtins.isString sparseCheckout then
+  throw "Please provide files/patterns for sparseCheckout as a list of strings, passing a (multi-line) string in is not supported anymore"
```

Comment:
Ideally, we'd warn here, but I think this is small enough that it's OK. (cc @mweinelt)

Reply:
Throwing here prevents it from erroring out with a less meaningful message in L87 I think.

Reply:
Warning, you mean? What I meant was warning but then making the newline string work as it did previously, by passing it directly.

Reply:
@SuperSandro2000's opinion from his earlier comment:

> The feature is relative new if I remember correct and had very few usages inside of nixpkgs. I think it is alright to hard error here.

Reply:
I would say that I prefer this being an error more, but I'm the least experienced person here.

Reply:
Updated to a emit a warning, yet still accept a string. Please review.